Remove the last of the Python 2 Code from the Project

cailafinn opened the pull request:
**Description of work.**
This PR should cover the removal of the last little bits of python 2 that were left from before, mainly removing if statements to check the py version in the C++ code and from the report generation scripts. 

**To test:**
There's a few different elements changed but nothing should be affected, mainly checking that TableWorkspaces can still be generated from scripts, that algorithms still work, and that unit labels can be changed from python. 

Fixes #27722 

*This does not require release notes* because **it is an internal change.**

**Note:** The parts of jenkins responsible for report generation may need updating after this is merged. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

Commit:
Remove PY_MAJOR_VERSION checks

We no longer support python 2, so these checks should be redundant.

RE #27722

Commit:
Remove old python2 code

These files are only used as part of system tests when running on the
build servers. Mostly for report generation

RE #27722

martyngigg commented:
Moving to 6.5 milestone as it is going to `main`.

martyngigg reviewed (changes requested):
All of the changes to existing files look good.

I think there may be an extra `failed.txt` file committed by accident?

Commit:
Remove compile error test file

Remove the file that originally had the list of files that didn't
compile in python3 that was erraneously committed.

RE #33880

martyngigg reviewed (approved):
Looks good! Thanks!
